api/v1alpha1/directivebreakdown_types.go

```diff
@@ -108,17 +117,32 @@ type StorageBreakdown struct {
 	AllocationSets []StorageAllocationSet `json:"allocationSets,omitempty"`
 }
 
+type ComputeLocationType string
+
+const (
+	ComputeLocationNetwork  ComputeLocationType = "network"
+	ComputeLocationPhysical ComputeLocationType = "physical"
+)
+
+type ComputeLocationPriority string
+
 const (
-	ComputeLocationNetwork  = "network"
-	ComputeLocationPhysical = "physical"
+	ComputeLocationPriorityMandatory  ComputeLocationPriority = "mandatory"
+	ComputeLocationPriorityBestEffort ComputeLocationPriority = "bestEffort"
 )
 
+type ComputeLocationAccess struct {
+	// Type is the relationship between the compute nodes and the resource in the Reference
+	// +kubebuilder:validation:Enum=physical;network
+	Type ComputeLocationType `json:"type"`
+
+	Priority ComputeLocationPriority `json:"priority"`
```

Comment:
This should have a comment, so the generated OpenAPI part has a description for it.